addons/test_mail/tests/test_message_track.py

```diff
@@ -545,22 +548,30 @@ def test_mail_track_all_no2many(self):
         new_message = test_record.message_ids - messages
         self.assertEqual(len(new_message), 1,
                          'Should have generated a tracking value')
-        self.assertTracking(
-            new_message,
-            [
-                ('boolean_field', 'boolean', 0, 1),
-                ('char_field', 'char', False, 'char_value'),
-                ('date_field', 'date', False, today_dt),
-                ('datetime_field', 'datetime', False, now),
-                ('float_field', 'float', 0, 3.22),
-                ('integer_field', 'integer', 0, 42),
-                ('many2one_field_id', 'many2one', self.env['res.partner'], self.test_partner),
-                ('monetary_field', 'monetary', False, (42.42, self.env.ref('base.USD'))),
-                ('selection_field', 'selection', '', 'FIRST'),
-                ('text_field', 'text', False, 'text_value'),
-            ],
-            strict=True
-        )
+        tracking_value_list = [
+            ('boolean_field', 'boolean', 0, 1),
+            ('char_field', 'char', False, 'char_value'),
+            ('date_field', 'date', False, today_dt),
+            ('datetime_field', 'datetime', False, now),
+            ('float_field', 'float', 0, 3.22),
+            ('float_field_with_digits', 'float', 0, 3.00001),
+            ('integer_field', 'integer', 0, 42),
+            ('many2one_field_id', 'many2one', self.env['res.partner'], self.test_partner),
+            ('monetary_field', 'monetary', False, (42.42, self.env.ref('base.USD'))),
+            ('selection_field', 'selection', '', 'FIRST'),
+            ('text_field', 'text', False, 'text_value'),
+        ]
+        self.assertTracking(new_message, tracking_value_list, strict=True)
+        # check formatting for all field types
+        formatted_values_all = new_message.message_format()[0]['trackingValues']
+        for (field_name, field_type, _, _), formatted_vals in zip(tracking_value_list, formatted_values_all):
+            currency = self.env.ref('base.USD').id if field_type == 'monetary' else False
+            precision = None if field_name != 'float_field_with_digits' else (10, 8)
+            with self.subTest(field_name=field_name):
+                self.assertEqual(formatted_vals['oldValue']['currencyId'], currency)
+                self.assertEqual(formatted_vals['newValue']['currencyId'], currency)
+                self.assertEqual(formatted_vals['oldValue']['floatPrecision'], precision)
+                self.assertEqual(formatted_vals['newValue']['floatPrecision'], precision)
```

Comment:
Is there a specific reason you need to test both old and new values? To me it looks like if the new value follows the correct scheme then the old value is the same.